cmd/serv.go

```diff
@@ -181,7 +181,14 @@ func runServ(c *cli.Context) error {
 
 	rr := strings.SplitN(repoPath, "/", 2)
 	if len(rr) != 2 {
-		return fail("Invalid repository path", "Invalid repository path: %v", repoPath)
+		if len(setting.SSH.RedirectPath) > 0 {
+			redirectPath := strings.ToLower(setting.SSH.RedirectPath)
```

Comment:
This is already being lowered on L194, it doesn't have to happen here already

Reply:
Thanks for looking at this.
I think we need to do lower on the setting.SSH.RedirectPath string. Since the resulting gitcmd, L301, L303, should get a lower case string to match the filesystem name?
But the ToLower on line L194 and L195 might not be needed because of the ToLower on L180?

Reply:
See other comment with suggestion.

Reply:
ok, I agree it is nicer to explicitly see the array with its elements rather than doing the split.
But building the new rr array we should use the old repoPath for the second element I think otherwise we might point to the wrong repo.  Something like the below?
```
	if len(rr) != 2 {
		if len(setting.SSH.RedirectPath) > 0 {
			redirectPath := strings.ToLower(setting.SSH.RedirectPath)
			rr = []string{redirectPath,repoPath)
			// build new repoPath used in gitcmd
			repoPath = redirectPath + "/" + repoPath
		} else {
			return fail("Invalid repository path", "Invalid repository path: %s", repoPath)
		}
	}
```

Do you think I should make another PR to remove the other redundant ToLower on rr[0] and rr[1] on L194,L195?
Then maybe we should reuse the already lowered setting.SSH.RedirectPath ?

Reply:
> ```
> 
> ```
> 
> 
>     
>   
> 
> Do you think I should make another PR to remove the other redundant ToLower on rr[0] and rr[1] on L194,L195?
> Then maybe we should reuse the already lowered setting.SSH.RedirectPath ?

They aren't redundant. A valid user/reponame would also need to be lowered.



> But building the new rr array we should use the old repoPath for the second element I think otherwise we might point to the wrong repo. Something like the below?

The only thing that differs here is that `rr` now contains the lowered RedirectPath, which is redunant as that's already done further down the codepath.

Reply:
The logic LGTM